main/ejml-core/src/org/ejml/masks/PrimitiveDMask.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2021, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ejml.masks;
+
+/**
+ * Mask implementation backed by a primitive array
+ */
+public class PrimitiveDMask extends Mask {
+
+    private final double[] values;
+    private final int numCols;
+    private final double zeroElement;
+
+    public PrimitiveDMask( double[] values, int numCols, boolean negated, double zeroElement ) {
+        // for dense structures they cannot be used for structural masks
+        super(negated);
+        this.values = values;
+        this.numCols = numCols;
+        this.zeroElement = zeroElement;
+    }
+
+    @Override
```

Comment:
Question: I've been experimenting with having `@Override` on the same line as the function. While not standard in java it reduces the number of vertical lines and doesn't make it less readable. Also fairly common in other languages. Thoughts?

Reply:
As its an annotation, I would keep it on a separate line. Out of curiosity, which languages handle override on the same line?

Reply:
C++ and Kotlin come to mind. In those cases it's part of the language. Java (probably for backwards compatibility) made it an optional annotation. When I was looking into code formatting I notice some formatters explicitly allow this style.

Reply:
Isn't the override als optional in C++? So it's a good question why override is an annotation in Java